src/passes/Precompute.cpp

```diff
@@ -134,9 +134,12 @@ struct Precompute
             curr->finalize();
             return;
           }
-        } else if (singleValue.isNull() &&
-                   curr->value->template is<RefNull>()) {
-          return;
+        } else if (singleValue.isNull()) {
+          if (auto* n = curr->value->template dynCast<RefNull>()) {
+            n->finalize(singleValue.type);
+            curr->finalize();
+            return;
+          }
```

Comment:
While on it I also looked at all the uses of `RefNull` again and found that this one might be problematic as well?